### ISSUE TITLE
IllegalArgumentException in FrequencyChart.drawColumn

### ISSUE THREAD

**Issue by iSoron** [iSoron/uhabits#967]
## Pre-submission checklist
- [x] I am submitting a bug report, not a feature request.
- [x] I have searched for similar issues, but did not find any matches.

## Description
From Google Play Console:

```
java.lang.IllegalArgumentException: 
  at org.isoron.uhabits.core.models.Timestamp.<init> (Timestamp.kt:127)
  at org.isoron.uhabits.core.models.Timestamp.<init> (Timestamp.kt:33)
  at org.isoron.uhabits.activities.common.views.FrequencyChart.drawColumn (FrequencyChart.kt:168)
  at org.isoron.uhabits.activities.common.views.FrequencyChart.onDraw (FrequencyChart.kt:134)
  at android.view.View.draw (View.java:22556)
  at android.view.View.draw (View.java:22429)
  at android.view.ViewGroup.drawChild (ViewGroup.java:4518)
  at android.view.ViewGroup.dispatchDraw (ViewGroup.java:4279)
  at android.view.View.draw (View.java:22427)
  at android.view.ViewGroup.drawChild (ViewGroup.java:4518)
  at android.view.ViewGroup.dispatchDraw (ViewGroup.java:4279)
  at android.view.View.draw (View.java:22559)
  at android.view.View.draw (View.java:22429)
  at android.view.ViewGroup.drawChild (ViewGroup.java:4518)
  at android.view.ViewGroup.dispatchDraw (ViewGroup.java:4279)
  at android.view.View.draw (View.java:22559)
  at org.isoron.uhabits.widgets.BaseWidget.getBitmapFromView (BaseWidget.kt:110)
  at org.isoron.uhabits.widgets.BaseWidget.buildRemoteViews (BaseWidget.kt:86)
  at org.isoron.uhabits.widgets.BaseWidget.getRemoteViews (BaseWidget.kt:120)
  at org.isoron.uhabits.widgets.BaseWidget.getLandscapeRemoteViews (BaseWidget.kt:48)
  at org.isoron.uhabits.widgets.StackRemoteViewsFactory.onDataSetChanged (StackRemoteViewsFactory.kt:145)
  at android.widget.RemoteViewsService$RemoteViewsFactoryAdapter.onDataSetChanged (RemoteViewsService.java:142)
  at com.android.internal.widget.IRemoteViewsFactory$Stub.onTransact (IRemoteViewsFactory.java:152)
  at android.os.Binder.execTransactInternal (Binder.java:1184)
  at android.os.Binder.execTransact (Binder.java:1143)
```

## Steps to reproduce
1. Create a Yes/No habit
2. Open the habit statistics screen
3. Scroll down to Frequency chart
4. Scroll back all the way to January 1, 1970
5. The app crashes

## System information
 - Phone: Google Pixel 3a
 - Phone Operating System: Android 11
 - App version: 2.0.2

**Comment by LordJon** [comment]
Probably already known but I can reproduce this bug in version 1.8.12 as well. It 'crashes' back to the main screen of the app.

**Comment by hiqua** [comment]
We don't support dates before 2000 do we? Given:
https://github.com/iSoron/uhabits/blob/dev/uhabits-core/src/jvmMain/java/org/isoron/uhabits/core/models/Timestamp.kt#L35-L39

So the solution is to prevent dates from going below 2000?

**Comment by iSoron** [comment]
@hiqua This has been partially fixed in branch `hotfix/2.0.3`

**Comment by iSoron** [comment]
Marking as `wont-fix`, since this will be automatically fixed when we migrate away from `org.isoron.uhabits.core.models.Timestamp` to the platform-independent `org.isoron.platform.time.LocalDate`. Charts that already use `LocalDate` are not affected by this bug.
